modules/SofaGeneralObjectInteraction/AttachConstraint.h

```diff
@@ -163,11 +166,9 @@ protected :
         }
         else
         {
-            Deriv in1 = dx1;
-            Deriv in2 = dx2;
-            Real constraintFactor = getConstraintFactor(index);
-            dx1 += in2*(d_responseFactor.getValue()*constraintFactor);
-            dx2 += in1*(d_responseFactor.getValue()*constraintFactor);
+            Deriv corr = (dx2-dx1)*0.5*d_responseFactor.getValue()*getConstraintFactor(index);
```

Comment:
Try to keep the data_field out of loops. Here the projectResponse function is called n-time...wile d_responseFactor is "assumed" constant during the time step. So pass the response factor as an argument.